java-core/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -767,4 +791,48 @@ public String getClientLibToken() {
   public String getQuotaProjectId() {
     return quotaProjectId;
   }
+
+  /**
+   * Returns the resolved endpoint for the Service to connect to Google Cloud
```

Comment:
I felt the term "host" in method name and "endpoint" in Javadoc are confusing.

Reply:
I pivoted to use host last night. I'll update the docs to reference host instead of endpoint. Host is used for Java-Core for backwards compatibility even though GAPICs use the endpoint terminology.

Reply:
Ensure this won't cause the discrepancy in user-facing interfaces.